### DISCUSSION TITLE
SDK 1.9 Notice type 06 - Missing schematron rules (BT-14-Part + GR-Part-EnvironLegis, GR-Part-EmployLegis, GR-Part-FiscalLegis)

### DISCUSSION THREAD

**Discussion by luispozoWS** [OP-TED/eForms-SDK#826] [Q&A] [answered]
Hello,
Using our own application we have two problems in notice type 06.
XSD errors appear in two cases despite no schematron rules indicating that these fields are mandatory:

No schematron rule indicates that BT-14-Part is mandatory.
![image](https://github.com/OP-TED/eForms-SDK/assets/158502029/1575fb36-5270-4742-83fd-6dd54c0ff0ad)

No schematron rule indicates that the following fields are also mandatory in groups GR-Part-EnvironLegis, GR-Part-EmployLegis and GR-Part-FiscalLegis 

- OPT-110-Part-FiscalLegis
- OPT-111-Part-FiscalLegis
- OPT-120-Part-EnvironLegis
- OPT-112-Part-EnvironLegis
- OPT-130-Part-EmployLegis
- OPT-113-Part-EmployLegis

![image](https://github.com/OP-TED/eForms-SDK/assets/158502029/220202d2-8cec-4c12-9be2-40262a2eadb9)

Thanks for your help

**Comment by YvesJo** [comment, marked as answer]
Hi,
The validation process starts with the lexical validation (i.e. validation against the XSD) and the schematron will only be evaluated after, therefore having schematron rules may not help. We may have as illustration [the case reported some days ago](https://github.com/OP-TED/eForms-SDK/discussions/818) where BT-14 is optional and and OPT-140-Lot forbidden when BT-14-Lot does not exist and Mandatory when it does; however when this rule is not reflected on the UI and/or the user does not fill in OPT-140-Lot before submission for validation while BT-14 exists, he will face an XSD error.
It could be possible to use the ID (OPT-111, OPT-112, OPT-113, OPT-140) as entry point and forbid the specification of further information (e.g. OPT-110, OPT-120, OPT-130, BT-14 ...) in its absence.
KR
